### DISCUSSION TITLE
Does the OPFS backend support timestamps?

### DISCUSSION THREAD

**Discussion by curiousdannii** [emscripten-core/emscripten#21443] [Q&A] [answered]
From what I can see of the OPFS API, it doesn't expose any timestamps for files. Were you somehow able to add support for them in the OPFS backend, or is it just a fundamental limitation of this API?

Edit: I missed that a FileSystemFileHandle can be turned into a File, which *does* have a `lastModified` timestamp. But I haven't confirmed whether it is actually set correctly in OPFS.

**Comment by tlively** [comment, marked as answer]
The WasmFS backend-independent code does keep track of timestamps for files, but they're initialized to the time the file is opened and are not persistent. Currently the backends have no visibility or control over them.

**Reply by curiousdannii** [reply]
Oh, interesting. Not even the Node backend uses real timestamps? And if there was an IndexedDB backend (I didn't realise there wasn't one yet), then you could store timestamps in it too.

Did you decide to just not persist timestamps because they can't be supported in OPFS? I.e., to keep things consistent between the backends.

**Reply by tlively** [reply]
It was mostly just to keep the backend API as simple as possible. I think it would be reasonable to accept patches to let backends opt-in to supporting their own timestamps, too.
